app/models/workflow.rb

```diff
@@ -110,6 +106,15 @@ def start(event)
 
   private
 
+  def self.copy_event_data_from source_event, target_events
+    target_events.each do |target_event|
+      target_event.data = Hash.new unless target_event.data
```

Comment:
Here's the key change...

If `.data` doesnt exist, we give it one. This was previously blowing up for certain event handler types.